frontend/lib/src/components/shared/BaseButton/styled-components.ts

```diff
@@ -332,6 +337,45 @@ export const StyledPillsButtonActive = styled(
   }
 })
 
+export const StyledSegmentedControlButton = styled(
+  StyledButtonGroupBaseButton
+)<RequiredBaseButtonProps>(({ theme }) => {
+  return {
+    padding: `${theme.spacing.twoXS} ${theme.spacing.lg}`,
+    borderRadius: theme.radii.none,
+    flex: "1 0 fit-content",
+    maxWidth: "100%",
+    marginRight: `-${theme.sizes.borderWidth}`, // Add negative margin to overlap borders
+
+    "&:first-child": {
+      borderTopLeftRadius: theme.radii.default,
+      borderBottomLeftRadius: theme.radii.default,
+    },
+    "&:last-child": {
+      borderTopRightRadius: theme.radii.default,
+      borderBottomRightRadius: theme.radii.default,
+      marginRight: theme.spacing.none, // Reset margin for the last child
+    },
+    "&:hover": {
+      zIndex: 100, // Make sure overlapped borders are visible
```

Comment:
this value should come from the `theme.zIndices` object. It would be best to use an existing zIndex, but if there is no suitable one, feel free to add a new. In context of the Advanced Theming project, we try to get rid of as many hard-coded values as possible 🙂

Reply:
maybe we can just use `zIndices.priority` here?

Reply:
@lukasmasuch I don't see `zIndices.priority` in [zIndices.ts](https://github.com/streamlit/streamlit/commit/6dc14aee21f9cba03e4b7781aa79c1a7c1550059#diff-40c0eb49b585146fd8542e5bc83a46075e9c4879055b6f60292d3e37992db75f). Were you recommending to add a new property?

Reply:
It is this one: https://github.com/streamlit/streamlit/blob/d47ec47654990ee199c951bafbd5006d33df2ae5/frontend/lib/src/theme/primitives/zIndices.ts#L39